src/stores/A_store.js

```diff
@@ -71,6 +71,53 @@ class A_store extends ReduceStore {
                 return {...state, stage, path, player: {x: nextLocation.x, y: nextLocation.y}};
             }
 
+            case ActionTypes.GENERATE_FOG: {
+                const {player, end} = state;
+                const {x, y} = player;
+                const revealedGrid = mapGenerator(player, end, new Grid(SIZE, SIZE));
+                const pathGrid = new Grid(SIZE, SIZE);
+
+                let visibleCells = revealedGrid.getVisible(x, y);
```

Comment:
okay, so, how about this
instead of doing a `getVisible()` twice, first on `revealedGrid` and second on `pathGrid`, we do it just once – on `pathGrid` since that's the final value we want to use anyway
then we change `obfuscateGrid()` to something like `copyCells()` which takes in a set of cells
so then `revealedGrid.copyCells(visibleCells)` just iterates over the set and copies over terrain/traversable stuff and we save on having to call `getVisible()` twice

Reply:
🐇